articles/flutter-vs-reactnative/index.md

```diff
@@ -0,0 +1,98 @@
+
+#### Take 2: A glorified webpage
+Mobile devices became more powerful, more compact, and more connected. They were getting more attention than our humble PC or laptops did. So companies were looking towards to developing for mobile while supporting the PC. Due to mobile being the future, but the PC market still having one of the largest user base. The thought process was to present a webpage wrapped in an app. Frameworks like ionic, PhoneGap, and Cordova tried to do it but the performance was not on par.
+
+#### Take 2.5 - Enter React Native
+React is a front end framework for the web. What React Native did was extend the functionality of React to native Android and iOS platforms. The developer got to code the UI as React components. The developer could use the same code used in their React web app for some UI elements. Later React Native would replace those web elements with the corresponding native elements. For example, ` <span> ` primitive is replaced with TextView in the native build and UI View in iOS build. This was better than the earlier approach because:
+1. Web developers could dive into mobile app development without a steep learning curve.
+2. Code can be reused to a large extent.
+3. It offered better performance than a web app wrapper as it ran closer to metal.
+
+#### Take 3: A Blank canvas
+Let's take a look at mobile game development frameworks like Unity. They don't have anything like the Native UI component. All they provide is a blank 2D canvas or an empty 3D space. You are free to show anything you want in that space, irrespective of the mobile platform you are on.
+
+Flutter works in the same way, unlike other app development frameworks. The platform is just concerned with rendering a canvas (Skia canvas). Once the canvas is created, Flutter paints on that canvas 60 times a second. That way it does not have to deal with platform-specific components. What prevents Flutter from supporting another platforms is its inability to run the Skia canvas on that platform. If you get that running, and you have another platform Flutter now supports, yay!
+
+#### What Flutter gets right
+##### Hot Reload-The MVP of Flutter
+Flutter hot reloads are super fast. Developers can make changes and view them on the fly within milliseconds. Want to try out which color suits the app? You can keep changing the color and it’ll show up in the app near instantly! This is great for rapid prototyping and creating an MVP.
+
+##### 1 codebase - multiple platforms
+Android, iOS, web, Windows, MacOS, Linux, smartwatches, IoT. That’s all the platforms that exist for users. (Painting on canvas, basically the Bob Ross of frameworks!)
+
+##### Smooth UI, rendered at 60fps
+Because Flutter is painting stuff, there's visual processing involved. This gets the GPU takes a generous load off the processors. So achieving 60 fps rendering is not a telling task for the framework.
```

Comment:
This sentence was  a bit difficult to read

Reply:
I have reworded it completely